service/src/main/java/org/databiosphere/workspacedataservice/InstanceInitializerBean.java

```
@@ -40,34 +49,89 @@ public boolean isInCloneMode(String sourceWorkspaceId) {
             try {
                 UUID.fromString(sourceWorkspaceId);
             } catch (IllegalArgumentException e){
-                    LOGGER.warn("Source workspace id could not be parsed, unable to clone DB. Provided source workspace id: {}", sourceWorkspaceId);
+                    LOGGER.warn("Source workspace Id could not be parsed, unable to clone DB. Provided source workspace id: {}.", sourceWorkspaceId);
                     return false;
             }
+
+            if (sourceWorkspaceId.equals(workspaceId)) {
+                LOGGER.warn("Source workspace id and current workspace Id can't be the same.");
+                return false;
```

Thank you for checking for this. Not something to resolve now but do you think we might need to do more to bubble up this error to users? Can't think of a way this would trigger in the wild but right now it's failing silently.

---

It's an interesting thought - if something like this happened, I am not sure what a user could do about it (and it could be confusing since user doesnt know how data tables work/etc in the background per say). It would likely be a more fundamental error somewhere in the terra infra, so I think at this point giving a user a blank WDS and saying cloning failed is likely an okay outcome. With that said, we can certainly talk this through with Beth as well to make sure the outcome makes sense from a product perspective. (i.e. would it make sense to surface this kind of error in the cloning status)